webxr/glwindow/mod.rs

```diff
@@ -380,3 +345,129 @@ impl GlWindowDevice {
         }
     }
 }
+
+struct GlWindowShader {
+    gl: Rc<dyn Gl>,
+    buffer: GLuint,
+    vao: GLuint,
+    program: GLuint,
+}
+
+const VERTEX_ATTRIBUTE: GLuint = 0;
+const VERTICES: &[[f32; 2]; 4] = &[[-1.0, -1.0], [-1.0, 1.0], [1.0, -1.0], [1.0, 1.0]];
+const VERTEX_SHADER: &[u8] = b"
+  #version 330 core
+  layout(location=0) in vec2 coord;
+  void main(void) {
+    gl_Position.xy = coord;
+    gl_Position.z = 0.0;
+    gl_Position.w = 1.0;
+  }
+";
+
+const FRAGMENT_SHADER: &[u8] = b"
+  #version 330 core
+  layout(location=0) out vec4 color;
+  uniform sampler2D image;
+  void main() {
+    ivec2 size = textureSize(image, 0);
+    vec2 position = vec2(gl_FragCoord.x / size.x, gl_FragCoord.y / size.y);
+    color = texture(image, position);
```

Comment:
This is fine, but I just use

    // vertex shader
    out vec2 vTexCoord;
    ...
    void main() {
        gl_Position = vec4(coord, 0.0, 1.0);
        vTexCoord = coord * 0.5 + 0.5;
    }

    // fragment shader
    ...
    in vec2 vTexCoord;
    void main() {
        color = texture(image, vTexCoord);
    }

Reply:
Done.